File: ps/zziri/leetcode/10. Regular Expression Matching.java
import java.util.regex.*;
class Solution {
    public boolean isMatch(String s, String p) {
        Pattern pattern = Pattern.compile(p);
        Matcher matcher = pattern.matcher(s);
        if(!matcher.matches()) {
            return false;
        }
        int start = matcher.start();
        int end = matcher.end();
        String temp = s.substring(start, end);
        if(s.equals(temp)) {
            return true;
        }
        else{
            return false;
        }
    }
}

File: zziri/ps/leetcode/Problems/10. Regular Expression Matching.py
import re
class Solution:
    def isMatch(self, s: str, p: str) -> bool:
        pat = re.compile(p)
        match = re.search(pat, s)
        if not match:
            return False
        else:
            if s == s[match.start(): match.end()]:
                return True
            else:
                return False

